### ISSUE TITLE
Custom linter: References should always be `const`

### ISSUE THREAD

**Issue by MonsterDruide1** [MonsterDruide1/OdysseyDecomp#70]
I haven't seen a non-const reference around anywhere yet, non-const objects are handled with pointers instead. So the linter should complain when a non-const `T&` is seen around, because it is most likely wrong and should either be a pointer `T*`, or `const T&`.

**Comment by LynxDev2** [comment]
While trying to fix this I noticed that there are a lot of non-const `auto&` in iterator for loops used in this project. Should this also apply to them?

**Comment by MonsterDruide1** [comment]
Good point - no, those should not be affected, as some of those require mutable references (e.g. `ByamlWriterBigDataList::setOffset`).

**Comment by LynxDev2** [comment]
Should I skip this check for all auto references since they include the constness from the return type or should I only skip it for iterator for loops?

**Comment by MonsterDruide1** [comment]
Do not skip it for anything else. The idea is that we want to make the `auto` more "verbose" by extracting those additional qualifiers (like `&`  and `const`). For loops, it *has* to be a non-const reference, so there is no other way to write it than `auto&` (or `auto`, but that violates the "more verbose" goal). Anything else that can take a const reference should also explicitly state `const`.
